<launch>
  <!-- Launch file for hesai32 LiDAR on the handheld rig in offline loc mode-->

    <arg name="rviz" default="true" />
    <arg name="configyaml" default="hesai32_mti3dk_handheld.yaml" />
    <arg name="bagfile" default="" />
    <arg name="tls_dir" default="" />
    <arg name="odom_mode" default="1" />
    <arg name="loc_follow_odom" default="0.0" />
    <arg name="init_lidar_pose_file" default="" />
    <arg name="msg_start_time" default="0" />
    <arg name="stationary_start" default="true" />
    <arg name="state_filename" default="scan_states.txt" />
    <arg name="save_dir" default="$(find fast_lio)/Log" />
    <arg name="tls_ref_traj_files" default="" />

    <rosparam command="load" file="$(find fast_lio)/config/$(arg configyaml)" />

    <param name="odom_mode" type="int" value="$(arg odom_mode)" />
    <param name="loc_follow_odom" type="double" value="$(arg loc_follow_odom)" />
    <param name="bagfile" type="string" value="$(arg bagfile)" />
    <param name="tls_dir" type="string" value="$(arg tls_dir)" />
    <param name="init_lidar_pose_file" type="string" value="$(arg init_lidar_pose_file)" />
    <param name="msg_start_time" type="double" value="$(arg msg_start_time)" />
    <param name="stationary_start" type="bool" value="$(arg stationary_start)" />
    <param name="state_filename" type="string" value="$(arg state_filename)" />
    <param name="tls_ref_traj_files" type="string" value="$(arg tls_ref_traj_files)" />
    <param name="feature_extract_enable" type="bool" value="0"/>
    <param name="point_filter_num" type="int" value="20"/>
    <param name="max_iteration" type="int" value="10" />
    <param name="scan_publish_enable" type="bool" value="1" />
	  <param name="dense_publish_enable" type="bool" value="1" />
    <param name="filter_size_surf" type="double" value="0.25" />
    <param name="filter_size_map" type="double" value="0.25" />
    <param name="cube_side_length" type="double" value="1000" />
    <param name="runtime_pos_log_enable" type="bool" value="0" />
    <param name="save_dir" type="string" value="$(arg save_dir)" />
    <node pkg="fast_lio" type="fastlio_mapping" name="laserMapping" output="screen" required="true" /> 

    <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="fastlio_rviz" args="-d $(find fast_lio)/rviz_cfg/loc.rviz" />
    </group>

</launch>